pkg/csi/service/mounter/mounter_windows.go

```diff
@@ -0,0 +1,475 @@
+	csiProxyMounter, err := newCSIProxyMounter(ctx)
+	if err == nil {
+		klog.V(2).Infof("using CSIProxyMounterV1, %s", csiProxyMounter.GetAPIVersions())
+		return &mount.SafeFormatAndMount{
+			Interface: csiProxyMounter,
+			Exec:      utilexec.New(),
+		}, nil
+	}
+	return nil, err
+}
+
+// newCSIProxyMounter - creates a new CSI Proxy mounter struct which encompassed all the
+// clients to the CSI proxy - filesystem, disk and volume clients.
+func newCSIProxyMounter(ctx context.Context) (*csiProxyMounter, error) {
+	fsClient, err := fsclient.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	diskClient, err := diskclient.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	volumeClient, err := volumeclient.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	return &csiProxyMounter{
+		FsClient:     fsClient,
+		DiskClient:   diskClient,
+		VolumeClient: volumeClient,
+		Ctx:          ctx,
+	}, nil
+}
+
+// GetAPIVersions returns the versions of the client APIs this mounter is using.
+func (mounter *csiProxyMounter) GetAPIVersions() string {
+	return fmt.Sprintf(
+		"API Versions filesystem: %s, disk: %s, volume: %s",
+		fsclient.Version,
+		diskclient.Version,
+		volumeclient.Version,
+	)
+}
+
+// normalizeWindowsPath normalizes windows path
+func normalizeWindowsPath(path string) string {
+	normalizedPath := strings.Replace(path, "/", "\\", -1)
+	if strings.HasPrefix(normalizedPath, "\\") {
+		normalizedPath = "c:" + normalizedPath
+	}
+	return normalizedPath
+}
+
+// ExistsPath - Checks if a path exists. This call does not perform follow link.
+func (mounter *csiProxyMounter) ExistsPath(path string) (bool, error) {
+	ctx := mounter.Ctx
+	log := logger.GetLogger(ctx)
+	isExistsResponse, err := mounter.FsClient.PathExists(context.Background(),
+		&fs.PathExistsRequest{
+			Path: normalizeWindowsPath(path),
+		})
+	if err != nil {
+		log.Errorf("Proxy returned error while checking if PathExists: %v", err)
+		return false, err
+	}
+	return isExistsResponse.Exists, err
+}
+
+// Rmdir - delete the given directory
+func (mounter *csiProxyMounter) Rmdir(path string) error {
+	rmdirRequest := &fs.RmdirRequest{
+		Path:  normalizeWindowsPath(path),
+		Force: true,
+	}
+	_, err := mounter.FsClient.Rmdir(context.Background(), rmdirRequest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// MakeDir - Creates a directory.
+func (mounter *csiProxyMounter) MakeDir(pathname string) error {
+	ctx := mounter.Ctx
+	log := logger.GetLogger(ctx)
+	mkdirReq := &fs.MkdirRequest{
+		Path: normalizeWindowsPath(pathname),
+	}
+	_, err := mounter.FsClient.Mkdir(context.Background(), mkdirReq)
+	if err != nil {
+		log.Infof("Error: %v", err)
+		return err
+	}
+	return nil
+}
+
+// Gets windows specific disk number from diskId
+func (mounter *csiProxyMounter) GetDiskNumber(diskID string) (string, error) {
+	ctx := mounter.Ctx
+	log := logger.GetLogger(ctx)
+	// Check device is attached
+
+	log.Debug("in getDiskNumber %s", diskID)
```

Comment:
Let's improve debug messages. Can we change 
`log.Debug("in getDiskNumber %s", diskID)` to `log.Debug("GetDiskNumber called with diskID: %q", diskID)`

Reply:
done